subscriber/src/database.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright 2018 Bitwise IO
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate postgres;
+use postgres::{Connection, TlsMode, Error};
+
+pub struct Database {
+    dsn: String,
+    conn: Option<Connection>
+}
+
+impl Database {
+    pub fn new(dsn: &str) -> Database {
+        Database {
+            dsn: dsn.to_string(),
+            conn: None
+        }
+    }
+
+    pub fn create_tables(&self) -> Result<(), Error>{
```

Comment:
I don't think the subscriber should be responsible for creating tables. For rust implementations we usually have a `*.sql` script initialize the database when the postgres container starts up